### ISSUE TITLE
SpringMultirabbit for auto-configuration of multiple ConnectionFactories

### ISSUE THREAD

**Issue by rwanderc** [spring-projects/spring-amqp#891]
## Enhancement

Recently I created a [blog post](https://inside.mytaxi.com/springboot-with-multiple-rabbitmq-brokers-cec203c3f77) about **SpringMultirabbit**. This is a solution to provide multiple ConnectionFactories, auto-configured, extending `RabbitProperties`, so as to have a simpler way to introduce different brokers in applications (e.g.: one for configuration hot-reloading and another for business events). This feature is available at http://github.com/mytaxi/spring-multirabbit

Out of the comments, @garyrussell suggested to contribute this work to the framework.
I have already followed the instructions at [Contributor Guidelines](https://github.com/spring-projects/spring-amqp/blob/master/CONTRIBUTING.adoc). However, couldn't find how to create the issue in Jira [AMQP project](https://jira.spring.io/secure/RapidBoard.jspa?rapidView=3&projectKey=AMQP&view=planning.nodetail), thus, I'm creating here in GitHub.

Before submitting the PRs, I wanted to know if there are any further instructions or concerns.

**Comment by garyrussell** [comment]
@rwanderc Sorry about that - we recently switched from JIRA to GitHub issues; there's a note on the JIRA home page, but we haven't updated the contributor guidelines yet.

My only thoughts/concerns is that some of the classes in the boot autoconfiguration part might belong here instead (`ConnectionFactoryContextWrapper` might be an example), so that non-boot users can use this functionality.

Obviously, the boot-specific parts will need to be a separate contribution over there, once this is merged. I haven't talked to the boot team yet, but I don't anticipate any problems with them accepting a PR.

I like the idea of auto-determining the admins but we might be able to make it simpler than using naming conventions, perhaps by adding a getter on the admin so we can match admin and container factory connection factories. That functionality can be merged into the main BPP rather than subclassing it.

Anyway, welcome; we look forward to working with you.

**Comment by rwanderc** [comment]
PR created https://github.com/spring-projects/spring-amqp/pull/1111

**Comment by garyrussell** [comment]
Nice PR# 😄
